Dockerfile

```diff
@@ -0,0 +1,29 @@
+FROM python:3.9.9-buster as base-image
+
+RUN apt-get update && apt-get install -y --no-install-recommends apt-utils \
+    bash \
+    curl \
+    && mkdir project
+WORKDIR /project
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+ENV PATH = "${PATH}:/root/.poetry/bin"
+COPY poetry.lock pyproject.toml ./
+RUN poetry config virtualenvs.create false --local && poetry install
```

Comment:
Nice optimisation, keeping this high up in the file (genuinely made my life playing about with this file much easier!)